src/react_native/gesture.cljs

```diff
@@ -76,3 +76,25 @@
   [gesture-flat-list (rn-flat-list/base-list-props props)])
 
 (def scroll-view (reagent/adapt-react-class ScrollView))
+
+;;; Custom gesture section-list
+(defn flatten-sections
+  [input-vector]
+  (reduce
```

Comment:
In terms of implementation, since we're flattening just one level, I tried to cook a solution with `mapcat`. As far as I checked, the implementation is equivalent, but should perform better since it's considerably simpler. Also `select-keys` is avoided, since that scans the entire map when we only need the title.

```clojure
(defn flatten-sections
  [sections]
  (mapcat (fn [{:keys [title data]}]
            (into [{:title title :header? true}] data))
          sections))
```

Here's the test, the equality below shows they're equivalent. The only difference is that the final result of `mapcat` is a lazy sequence, whereas the `reduce` you used returns a vector. But since the result of `flatten-sections` is passed down to `flat-list`, it doesn't matter if it's a vector or not.

```clojure
(let [sections
      [{:title "Section A",
        :data
        [{:primary-name "Alice"
          :public-key   "0x1"}
         {:primary-name "Bob"
          :public-key   "0x2"}]}
       {:title "Section B",
        :data
        [{:primary-name "Carol"
          :public-key   "0x3"}
         {:primary-name "Daniel"
          :public-key   "0x4"}]}]]
  (= (flatten-sections-old sections)
     (flatten-sections-new sections)))

;; Result

;; => ({:title "Section A", :header? true}
;;     {:primary-name "Alice", :public-key "0x1"}
;;     {:primary-name "Bob", :public-key "0x2"}
;;     {:title "Section B", :header? true}
;;     {:primary-name "Carol", :public-key "0x3"}
;;     {:primary-name "Daniel", :public-key "0x4"})
```

Reply:
Thanks @ilmotta ! This one definitely looks simpler